source/s3_request_messages.c

```diff
@@ -353,6 +384,77 @@ struct aws_input_stream *aws_s3_message_util_assign_body(
     return NULL;
 }
 
+/* Add a content-range header.*/
+static int s_s3_message_util_add_content_range_header(
+    uint64_t part_index,
+    uint64_t part_size,
+    struct aws_http_message *out_message) {
+    AWS_PRECONDITION(out_message);
+
+    uint64_t range_start = part_index * part_size;
+    uint64_t range_end = range_start + part_size - 1;
+
+    /* TODO this is more than enough space, but maybe there's a better way to do this?
+     * ((2^64)-1 = 20 characters;  2*20 + length-of("bytes=-") < 128) */
+    char range_value_buffer[128] = "";
```

Comment:
We have a byte buffer append Api which is safer

Reply:
I guess we could use appends for some of the string parts, but I think we still need to access the buffer directly for the uint64 -> string formatting?